src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -1028,6 +1031,9 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 	}
 
 	isReadonly(): boolean {
+		if (this.lastResolvedFileStat?.readonly) {
```

Comment:
This is wrong, no? I think if we have a `boolean` `readonly` property we have to respect that independent from the file system? Because someone might contribute a readonly file system where selected files are not readonly?

Reply:
Fair point. But if someone is already contributing a readonly filesystem whose `stat` doesn't return any value for the file-level `readonly` property (because it didn't exist), we must default it to `true` for a readonly FS and `false` otherwise, right?

Reply:
Yeah, only take it unless `undefined`

Reply:
Please review the change I made in latest push.